R/value-box.R

```diff
@@ -85,90 +168,128 @@ value_box_dependency_sass <- function(theme) {
+  width_full_screen = "1fr"
+) {
+  new_showcase_layout(
+    class = "showcase-top-right",
+    width = width,
+    width_full_screen = width_full_screen,
+    max_height = max_height,
+    max_height_full_screen = max_height_full_screen
+  )
 }
 
+new_showcase_layout <- function(
+  class,
+  width = 0.3,
+  width_full_screen = "1fr",
+  max_height = "100px",
+  max_height_full_screen = 0.67
+) {
+  width <- validate_grid_width_unit(width)
+  width_full_screen <- validate_grid_width_unit(width_full_screen)
 
-showcase_layout_ <- function(width, max_height, max_height_full_screen, top_right) {
-
-  width <- validate_width_unit(width)
   max_height <- validate_height_unit(max_height)
   max_height_full_screen <- validate_height_unit(max_height_full_screen)
 
-  function(showcase, contents) {
-
-    showcase_container <- div(
-      class = "value-box-showcase",
-      class = "overflow-hidden",
-      class = if (top_right) "showcase-top-right",
-      style = css(
-        "--bslib-value-box-max-height" = max_height,
-        "--bslib-value-box-max-height-full-screen" = max_height_full_screen
-      ),
-      showcase
-    )
-
-    showcase_container <- bindFillRole(showcase_container, container = TRUE, item = TRUE)
+  structure(
+    list(
+      class = class,
+      width = width,
+      width_full_screen = width_full_screen,
+      max_height = max_height,
+      max_height_full_screen = max_height_full_screen
+    ),
+    class = "bslib_showcase_layout"
+  )
+}
 
-    if (!top_right) {
-      contents <- tagAppendAttributes(contents, class = "border-start")
-    }
+#' @export
+print.bslib_showcase_layout <- function(x, ...) {
+  cat("<", x$class, ">\n", sep = "")
+  cat("width: ", x$width, " [fs: ", x$width_full_screen, "]\n", sep = "")
+  cat("max_height:", x$max_height, " [fs: ", x$max_height_full_screen, "]\n", sep = "")
+  invisible(x)
+}
 
-    items <- list(showcase_container, contents)
-    width_fs <- c("1fr", "auto")
-    if (top_right) {
-      items <- rev(items)
-      width_fs <- rev(width_fs)
-    }
+layout_showcase <- function(showcase_layout, showcase, contents) {
+  showcase_container <- div(
+    class = "value-box-showcase",
+    style = css(
+      "---bslib-value-box-showcase-max-h" = showcase_layout$max_height,
+      "---bslib-value-box-showcase-max-h-fs" = showcase_layout$max_height_full_screen
+    ),
+    showcase
+  )
 
-    card_body(
-      style = css(padding = 0),
-      layout_column_wrap(
-        width = NULL, gap = 0,
-        heights_equal = "row",
-        class = "value-box-grid",
-        style = css(
-          "--bslib-value-box-widths" = width,
-          "--bslib-value-box-widths-full-screen" = width_fs
-        ),
-        !!!items
-      )
+  card_body(
+    style = css(padding = 0),
+    layout_column_wrap(
+      width = NULL, gap = 0,
+      heights_equal = "row",
+      class = "value-box-grid",
+      style = css(
+        "---bslib-value-box-showcase-w" = showcase_layout$width,
```

Comment:
Let's add a comment (maybe not here but in the Sass?) about how these are "private".

Also please file an issue about `htmltools::css()`